secretsclient/client.go

```diff
@@ -121,17 +121,17 @@ func (c *client) doCall(method, url string, opts ...requestOpt) (int, string, er
 	duration := time.Millisecond * 100
 	for attempt := 0; attempt < maxAttempt; attempt++ {
 		status, body, err = c.doCallImp(r, method, url, opts...)
-		if err == nil && status < 500 {
+		if (err == nil && status < 500) || err == ErrNoAuthorizedPublicKeys || err == ErrUnauthorized {
```

Comment:
Aren't these 4xx errors already?

Reply:
the ErrNoAuthorizedPublicKeys occurs once https://github.com/earthly/earthly/blob/main/secretsclient/client.go#L398-L408 exhausts all potential keys to try authenticating with.

Reply:
ErrUnauthorized isn't needed here, I can remove that.